File: yt/yt/flow/examples/java/async_request/async_request/src/main/java/tech/ytsaurus/flow/examples/asyncrequest/AsyncRequestComputationContext.java
package tech.ytsaurus.flow.examples.asyncrequest;

import java.util.List;

import org.springframework.context.annotation.Configuration;
import tech.ytsaurus.flow.computation.Computation;
import tech.ytsaurus.flow.spring.ComputationProvider;

// [BEGIN computation_context]
@Configuration
public class AsyncRequestComputationContext implements ComputationProvider {

    @Override
    public List<Computation> getComputations() {
        return List.of(
                Computation.builder()
                        .setComputationId("state")
                        .setProcessFunction(new StateKeeperFunction())
                        .build(),
                Computation.builder()
                        .setComputationId("processor")
                        .setProcessFunction(new RequestProcessorFunction())
                        .build()
        );
    }
}
// [END computation_context]

File: yt/yt/flow/examples/kotlin/async_request/async_request/src/main/kotlin/tech/ytsaurus/flow/examples/asyncrequest/AsyncRequestComputationContext.kt
package tech.ytsaurus.flow.examples.asyncrequest

import org.springframework.context.annotation.Configuration
import tech.ytsaurus.flow.computation.Computation
import tech.ytsaurus.flow.spring.ComputationProvider

// [BEGIN computation_context]
@Configuration
class AsyncRequestComputationContext : ComputationProvider {

    override fun getComputations(): List<Computation> {
        return listOf(
            Computation.builder()
                .setComputationId("state")
                .setProcessFunction(StateKeeperFunction())
                .build(),
            Computation.builder()
                .setComputationId("processor")
                .setProcessFunction(RequestProcessorFunction())
                .build(),
        )
    }
}
// [END computation_context]
